database/engine/rrdengine.c

```diff
@@ -54,813 +53,1766 @@ static void sanity_check(void)
+                        __func__, epdl->extent_offset, epdl->extent_size, epdl->datafile->fileno, i, count);
+            continue;
+        }
+        mrg_metric_release(main_mrg, metric);
+
+        if(worker)
+            worker_is_busy(UV_EVENT_PAGE_LOOKUP);
+
+        struct page_details *pd = NULL;
+        Pvoid_t *pd_by_start_time_s_judyL = JudyLGet(epdl->page_details_by_metric_id_JudyL, metric_id, PJE0);
+        internal_fatal(pd_by_start_time_s_judyL == PJERR, "DBENGINE: corrupted extent metrics JudyL");
+
+        if(pd_by_start_time_s_judyL && *pd_by_start_time_s_judyL) {
+            Pvoid_t *pd_pptr = JudyLGet(*pd_by_start_time_s_judyL, start_time_s, PJE0);
+            internal_fatal(pd_pptr == PJERR, "DBENGINE: corrupted metric page details JudyHS");
+
+            if(pd_pptr && *pd_pptr) {
+                pd = *pd_pptr;
+                internal_fatal(metric_id != pd->metric_id, "DBENGINE: metric ids do not match");
+            }
+        }
+
+        if(!pd && !preload_all_pages)
+            continue;
+
+        VALIDATED_PAGE_DESCRIPTOR vd = validate_extent_page_descr(
+                &header->descr[i], now_s,
+                (pd) ? pd->update_every_s : 0,
+                have_read_error);
+
+        if(worker)
+            worker_is_busy(UV_EVENT_PAGE_POPULATION);
+
+        PGC_PAGE *page = pgc_page_get_and_acquire(main_cache, (Word_t)ctx, metric_id, start_time_s, PGC_SEARCH_EXACT);
+        if (!page) {
+            void *page_data = dbengine_page_alloc(ctx, vd.page_length);
+
+            if (unlikely(!vd.data_on_disk_valid)) {
+                fill_page_with_nulls(page_data, vd.page_length, vd.type);
+                stats_load_invalid_page++;
+            }
+
+            else if (RRD_NO_COMPRESSION == header->compression_algorithm) {
+                memcpy(page_data, data + payload_offset + page_offset, (size_t) vd.page_length);
+                stats_load_uncompressed++;
+            }
+
+            else {
+                if(unlikely(page_offset + vd.page_length > uncompressed_payload_length)) {
+                    error_limit_static_global_var(erl, 10, 0);
+                    error_limit(&erl,
+                                   "DBENGINE: page %u offset %u + page length %zu exceeds the uncompressed buffer size %u",
+                                   i, page_offset, vd.page_length, uncompressed_payload_length);
+
+                    fill_page_with_nulls(page_data, vd.page_length, vd.type);
+                    stats_load_invalid_page++;
+                }
+                else {
+                    memcpy(page_data, uncompressed_buf + page_offset, vd.page_length);
+                    stats_load_compressed++;
+                }
+            }
+
+            PGC_ENTRY page_entry = {
+                .hot = false,
+                .section = (Word_t)ctx,
+                .metric_id = metric_id,
+                .start_time_t = vd.start_time_s,
```

Comment:
```sh
/database/engine/rrdengine.c: 1586 in extent_uncompress_and_populate_pages()
1580                 PGC_ENTRY page_entry = {
1581                     .hot = false,
1582                     .section = (Word_t)ctx,
1583                     .metric_id = metric_id,
1584                     .start_time_t = vd.start_time_s,
1585                     .end_time_t = vd.end_time_s,
>>>     CID 433481:  High impact quality  (Y2K38_SAFETY)
>>>     A "time_t" value is stored in an integer with too few bits to accommodate it.  The expression "vd.update_every_s" is cast to "uint32_t".
1586                     .update_every = vd.update_every_s,
1587                     .size = (size_t) vd.page_length,
1588                     .data = page_data
1589                 };
1590     
1591                 bool added = true;
```

I think these errors were already present in different PRs. :thinking:

Reply:
update every again